benchexec/tools/bubaak.py

```diff
@@ -0,0 +1,65 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+# SPDX-FileCopyrightText: 2022 Marek Chalupa marek.chalupa@ista.ac.at
+#
+# SPDX-License-Identifier: Apache-2.0
+
+""
+
+import benchexec
+import benchexec.result as result
+import benchexec.util as util
+from benchexec.tools.sv_benchmarks_util import get_data_model_from_task, ILP32, LP64
+
+def get_verdict(s: str):
+    return s[17:].strip()
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    Info object for the tool Bubaak.
+    """
+
+    REQUIRED_PATHS = ["."]
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("bubaak")
+
+    def version(self, executable):
+        return self._version_from_tool(executable, arg="-version")
+
+    def name(self):
+        """ The human-readable name of the tool. """
+        return "Bubaak"
+
+    def cmdline(self, executable, options, task, rlimits):
+        if task.property_file:
+            options += ["-prp", task.property_file]
+
+        arch = get_data_model_from_task(task, {ILP32: "-32", LP64: "-64"})
+        if arch is not None and arch not in options:
+            options += [arch]
+
+        return [executable, "-sv-comp"]\
+               + options\
+               + list(task.input_files_or_identifier)
+
+    def determine_result(self, run):
+        if run.was_timeout:
+            return "timeout"
```

Comment:
This check is not required, BenchExec will do that anyway. But you can keep it if you want to avoid executing the rest of the method in case of a timeout. In this case, please return `"TIMEOUT"`, though.